feat(lng): Added change language on the fly

priscilawebdev opened the pull request:
Change Language on the fly

![image](https://user-images.githubusercontent.com/29228205/79699590-bb730400-8290-11ea-9a8b-b9eabe0d215b.png)

![image](https://user-images.githubusercontent.com/29228205/79699594-c463d580-8290-11ea-94c7-b6cb91d0b980.png)

juanpicado commented on src/components/Icon/Icon.tsx:
```diff
@@ -11,6 +11,7 @@ import nicaragua from './img/nicaragua.svg';
 import pakistan from './img/pakistan.svg';
 import austria from './img/austria.svg';
 import spain from './img/spain.svg';
+import usa from './img/usa.svg';
```
USA? 🙃

priscilawebdev replied:
yes :) United States of America

juanpicado commented on src/components/LanguageSwitch/LanguageSwitch.tsx:
```diff
@@ -0,0 +1,157 @@
+
+  const { translation: userLanguage } = getTranslatedCurrentLanguage(t)[currentLanguage.toLowerCase()];
+
+  const handleToggle = () => {
```
we might optimize the callbacks with `useCallback`, what do you think?

priscilawebdev replied:
I think it's valid :)

juanpicado commented on src/components/LanguageSwitch/LanguageSwitch.tsx:
```diff
@@ -0,0 +1,157 @@
+                    <Divider />
+                  </Box>
+                  <MenuItem button={true}>
+                    <StyledLink external={true} onClick={handleClose} to="https://github.com/verdaccio/ui#translations">
```
I'd add this to a constant above, it is hardly visible here `https://github.com/verdaccio/ui#translations` . Furthermore, the README might change and this link will be broken. I'd rather remove the `#translations` part to be sure.

juanpicado commented:
I just merged French, I think we need also to update it here.

priscilawebdev commented:
@juanpicado thanks for the nice feedback! I just added the current language to localStorage, so it should look good now if you reload the page ...

I also updated the menu background in dark mode 😉

Regarding the header on mobile devices, I plan to refactor it by adding a mobile menu, because atm it is impossible to add another icon / information ... as you said, it looks odd ... I will do this in a follow-up PR ... but in the meantime, the component switchLanguage will be hidden on smaller devices 

I'm not having much time lately, but I will try to do that as soon as I have some free time :)

juanpicado commented:
> @juanpicado thanks for the nice feedback! I just added the current language to localStorage, so it should be fine now if you reload the page ...
> 
> I also updated the menu background color in dark mode 😉
> 
> Regarding the header on mobile devices, I plan to refactor it by adding a mobile menu, because atm it is impossible to add another icon / information ... as you said, it looks odd ... I will do this in a follow-up PR ... but in the meantime, the component switchLanguage will be hidden on smaller devices
> 
> I'm not having much time lately, but I will try to do that as soon as I have some free time 🙂

Perfect, good solution 👍
